src/GafferScene/TransformQuery.cpp

```diff
@@ -0,0 +1,522 @@
+            if( ! loc.empty() )
+            {
+                ScenePlug const* const p_scene_plug = scenePlug();
+                assert( p_scene_plug != 0 );
+
+                ScenePlug::ScenePath const path = ScenePlug::stringToPath( loc );
+
+                if( p_scene_plug->exists( path ) )
+                {
+                    switch( space() )
+                    {
+                        case SpaceLocal:
+                        {
+                            io_hash = p_scene_plug->transformHash( path );
+
+                            if( invertPlug()->getValue() )
+                            {
+                                invertPlug()->hash( io_hash );
+                            }
+                            break;
+                        }
+                        case SpaceWorld:
+                        {
+                            io_hash = p_scene_plug->fullTransformHash( path );
+                            
+                            if( invertPlug()->getValue() )
+                            {
+                                invertPlug()->hash( io_hash );
+                            }
+                            break;
+                        }
+                        case SpaceRelative:
+                        {
+                            std::string const rloc = relativeLocationPlug()->getValue();
+
+                            if( ! rloc.empty() && ( rloc != loc ) ) // NOTE : If loc == rloc then m is identity
+                            {
+                                io_hash.append( p_scene_plug->fullTransformHash( path ) );
+                                ScenePlug::ScenePath const rpath = ScenePlug::stringToPath( rloc );
+                                io_hash.append( p_scene_plug->fullTransformHash( rpath ) );
+                                invertPlug()->hash( io_hash );
+                            }
+                            break;
+                        }
+                        default:
+                            break;
+                    }
+                }
+            }
+        }
+        else
+        if( ip_plug != 0 )
+        {
+            Gaffer::GraphComponent const* const p_plug_parent = ip_plug->parent();
+
+            if( p_plug_parent == outTranslatePlug() )
+            {
+                ComputeNode::hash( ip_plug, ip_context, io_hash );
+                outMatrixPlug()->hash( io_hash );
+            }
+            else
+            if( p_plug_parent == outRotatePlug() )
+            {
+                ComputeNode::hash( ip_plug, ip_context, io_hash );
+                outMatrixPlug()->hash( io_hash );
+                rotateOrderPlug()->hash( io_hash );
+            }
+            else
+            if( p_plug_parent == outScalePlug() )
+            {
+                ComputeNode::hash( ip_plug, ip_context, io_hash );
+                outMatrixPlug()->hash( io_hash );
+            }
+        }
```

Comment:
If we were ever to be used as a base class, we would need to call `ComputeNode::hash()` for the case where the output plug isn't one of ours. When not trying to do hash-pass-throughs this is easy, because then you only need a single `ComputeNode::hash()` call at the top of the function, that is used for all output plugs.